homeassistant/components/slide/__init__.py

```diff
@@ -0,0 +1,159 @@
+"""Component for the Go Slide API."""
+
+import logging
+from datetime import timedelta
+
+import voluptuous as vol
+
+from homeassistant.const import (
+    CONF_USERNAME,
+    CONF_PASSWORD,
+    CONF_SCAN_INTERVAL,
+    CONF_TIMEOUT,
+)
+from homeassistant.helpers import config_validation as cv
+from homeassistant.helpers.discovery import async_load_platform
+from homeassistant.helpers.event import async_track_time_interval, async_call_later
+from .const import (
+    DOMAIN,
+    SLIDES,
+    API,
+    COMPONENT,
+    DEFAULT_TIMEOUT,
+    CONF_RETRY,
+    DEFAULT_RETRY,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_SCAN_INTERVAL = timedelta(seconds=30)
+
+CONFIG_SCHEMA = vol.Schema(
+    {
+        DOMAIN: vol.Schema(
+            {
+                vol.Required(CONF_USERNAME): cv.string,
+                vol.Required(CONF_PASSWORD): cv.string,
+                vol.Optional(
+                    CONF_SCAN_INTERVAL, default=DEFAULT_SCAN_INTERVAL
+                ): cv.time_period,
+                vol.Optional(CONF_TIMEOUT, default=DEFAULT_TIMEOUT): cv.positive_int,
+                vol.Optional(CONF_RETRY, default=DEFAULT_RETRY): cv.positive_int,
```

Comment:
Why do we need a config option for retry?

Reply:
The retry is used when an exception is caught during setup, thus the seconds between we should retry the setup of the 'slide' component again. The exception are normally connection errors, so not authentication issues. When authentication fails, the setup will stop and will not retry.

Reply:
And why does it need to be configurable?

Reply:
That's fine with me too, i will remove this in the next commit.